src/UI/Art/SpriteArtist/DamageChartView.java

```diff
@@ -0,0 +1,220 @@
+package UI.Art.SpriteArtist;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+
+import Engine.IView;
+import UI.DamageChartController;
+import UI.SlidingValue;
+import UI.UIUtils;
+import UI.UIUtils.Faction;
+import Units.UnitModel;
+import Units.WeaponModel;
+
+public class DamageChartView implements IView
+{
+  // Keep track of all the unit images we need to juggle.
+  private BufferedImage[] unitArray;
+  private final int unitSizePx; // Units are square
+  private final int unitBuffer;
+  private final int unitSpacingH;
+  private final int unitSpacingV;
+  private final int gridWidth;
+  private final int gridHeight;
+
+  private DamageChartController control;
+  private SlidingValue viewX = new SlidingValue(0);
+  private SlidingValue viewY = new SlidingValue(0);
+  private SpriteCursor spriteCursor = new SpriteCursor(0, 0, SpriteLibrary.baseSpriteSize, SpriteLibrary.baseSpriteSize, UIUtils.getCOColors()[0]);
+
+  public DamageChartView(DamageChartController control)
+  {
+    this.control = control;
+    unitArray = new BufferedImage[control.units.length];
+
+    // Just pull one sprite to start with. Loading everything at once might take a while.
+    unitArray[0] = SpriteLibrary.getMapUnitSpriteSet(control.units[0].name, control.shooters, control.shooterColor).sprites[0].getFrame(0);
```

Comment:
Did you actually see lag trying to do all at once? Presumably the unit images should already be loaded by the time we get here anyway.

Reply:
Copy/pasted comment, to be fair.
Also, this view is guaranteed to show all unit types (including sea/air), which aren't guaranteed to ever be loaded otherwise.

Reply:
Every unit type, yes, but in only two colors.